Assets/Shopify/Plugins/iOS/Shopify/PaymentAuthorizationControlling/PaymentAuthorizationControlling.swift

```diff
@@ -31,7 +31,7 @@ import PassKit
     
     var authorizationDelegate: NSObjectProtocol? { get set }
     
-    init(paymentRequest request: PKPaymentRequest)
+    init?(paymentRequest request: PKPaymentRequest)
```

Comment:
Why is this necessary? Making this a failable initializer effectively cascades that dependency down to `PaymentSession` where it is force-unwrapped.

Reply:
Looks like this protocol is used both on `PKPaymentAuthorizationController` and `PKPaymentAuthorizationViewController`. The first has a regular initializer while the view controller class has a failable one and Swift 4 throws an error saying that the protocol doesn't support overriding a failable initialier with a normal one. I guess making this failable is more true of what could happen if `PaymentSession` is nil. In a previous commit I did the force unwrap (bad) but I've since updated it to properly return a `false` indicating that that the user has no supported payment options.

Reply:
Sounds reasonable 👍